content/en/blog/_posts/2023-09-15-Bootstrap-an-Air-Gapped-Cluster-with-Kubeadm/index.md

```diff
@@ -0,0 +1,665 @@
+    "registry.k8s.io/pause:3.9"
+    "registry.k8s.io/etcd:3.5.7-0"
+    "registry.k8s.io/coredns/coredns:v1.10.1"
+    "registry:2.8.2"
+    "flannel/flannel:v0.22.0"
+    "flannel/flannel-cni-plugin:v1.1.2"
+)
+
+for image in "${images[@]}"; do
+    # Pull the image from the registry
+    docker pull "$image"
+
+    # Save the image to a tar file on the local disk
+    image_name=$(echo "$image" | sed 's|/|_|g' | sed 's/:/_/g')
+    docker save -o "${image_name}.tar" "$image"
+
+done
+```
+
+The above commands will take a look at the CPU architecture for the current host/laptop, create and change into a directory called download, and finally download all of the dependencies. Each of these files must then be transported over the air gap via scp. The exact syntax of the command will vary depending on the user on the VM, if you created an SSH key, and the IP of your air gap VM. The rough syntax is:
+```bash
+scp -i <<SSH_KEY>> <<FILE>> <<AIRGAP_VM_USER>>@<<AIRGAP_VM_IP>>:~/tmp/
+```
+Once all of the files have been transported to the air gapped VM, the rest of the blog post will take place from the VM. Open a terminal session to that system.
+
+**Put the artifacts in place**
+
+Everything that is needed in order to bootstrap a Kubernetes cluster now exists on the air-gapped VM. This section is a lot more complicated since various types of artifacts are now on disk on the air-gapped VM. Get a root shell on the air gap VM as the rest of this section will be executed from there. Let's start by setting the same architecture variables and environmental as were set on the host/laptop and then install all of the RPM packages:
+
+```bash
+UARCH=$(uname -m)
+# Set architecture variables
+
+if [["$UARCH" == "arm64" || "$UARCH" == "aarch64"]]; then
+
+    ARCH="aarch64"
+    K8s_ARCH="arm64"
+
+else
+
+    ARCH="x86_64"
+    K8s_ARCH="amd64"
+
+fi
+
+# Set environmental vartiables
+CNI_PLUGINS_VERSION="v1.3.0"
+CRICTL_VERSION="v1.27.0"
+KUBE_RELEASE="v1.27.3"
+RELEASE_VERSION="v0.15.1"
+K9S_VERSION="v0.27.4"
+
+cd ~/tmp/
+
+dnf -y install ./*.rpm
+```
+
+**Next, install the CNI plugins and crictl:**
+
+
+```bash
+mkdir -p /opt/cni/bin
+tar -C /opt/cni/bin -xz -f "cni-plugins-linux-${K8s_ARCH}-v1.3.0.tgz"
+tar -C /usr/local/bin-xz -f "crictl-v1.27.0-linux-${K8s_ARCH}.tar.gz"
+```
+
+**Make kubeadm, kubelet and kubectl executable and move them from the /tmp directory to /usr/local/bin:**
+```bash
+chmod +x kubeadm kubelet kubectl
+mv kubeadm kubelet kubectl /usr/local/bin
+```
+**Define an override for the systemd kubelet service file and move it to the proper location.**
+
+```bash
+mkdir -p /etc/systemd/system/kubelet.service.d
+
+sed "s:/usr/bin:/usr/local/bin:g" 10-kubeadm.conf > /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+```
+
+**The CRI plugin for containerd is disabled by default; enable it:**
+```bash
+sed -i 's/^disabled_plugins = \["cri"\]/#&/' /etc/containerd/config.toml
+```
+**Put the /etc/docker/daemon.json file in place:**
+```bash
```

Comment:
Check line endings here.

Reply:
@sftim , apologies. What line endings should I be looking at here?